src/HealthSystemKernel.java

```diff
@@ -0,0 +1,50 @@
+import components.standard.Standard;
+
+/**
+ * Health system kernel component with primary methods for managing character's
+ * health.
+ *
+ * @mathmodel type HealthSystemKernel is modeled by the number of health points
+ *            (HP).
+ * @initially ensures this = MAX_HEALTH (character starts with full health)
+ */
+public interface HealthSystemKernel extends Standard<HealthSystemKernel> {
+
+    /**
+     * Decreases the health by the specified damage amount.
+     *
+     * @param damage
+     *            the amount of damage taken
+     * @updates this
+     * @requires damage >= 0
+     * @ensures this = #this - damage
+     */
+    void takeDamage(int damage);
```

Comment:
I like these method names for clarity. I might combine them since in my mind damage is just negative healing. These could maybe be combined into a method like `adjustHP`